base/src/com/google/idea/blaze/base/dependencies/BlazeQuerySourceToTargetProvider.java

```diff
@@ -201,7 +210,27 @@ private static ImmutableList<TargetInfo> getTargetInfoList(
       return blazeQueryLabelKindParser.getTargets();
     } catch (IOException e) {
       throw new BlazeQuerySourceToTargetException("Failed to get target info list", e);
+    } finally {
+      if (!Registry.get("bazel.sync.keep.query.files").asBoolean()) {
+        queryFile.toFile().delete();
+      }
+    }
+  }
+
+  private static @NotNull Path prepareQueryFile(Project project, String rdepsQuery) throws BlazeQuerySourceToTargetException {
+    Path queryFile = null;
+    try {
+      Path queriesDir = Paths.get(project.getBasePath()).resolve("queries");
```

Comment:
I suspect this should be store in tmpdir and not somewhere inside the project

Reply:
I think it's a good practice to keep it where all internal files are. In such a case the directory permisisons are the same as the rest of the project

Reply:
ok let's keep in inside the project at least for now